draft-jhoyla-tls-key-injection.md

```diff
@@ -53,27 +53,25 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
-# PSK Injection
+# Key Schedule Injection
+
+This section describes ways in which additional secrets can be injected into
+the TLS 1.3 key schedule.
+
+## Early Secret Injection
 
 TLS provides exporter keys that allow for other protocols to provide
 data authenticated by the TLS channel. This can be used to bind a protocol to a
 specific TLS handshake, giving joint authentication guarantees.
+In a similar way, one may wish to introduce externally authenticated and 
+pre-shared data to the early secret derivation. This can be used to bind external 
+protocols to the TLS protocol. 
 
-In a similar way, this mechanism describes a way to introduce externally
-authenticated data to a TLS handshake.  To add additional authenticated data
-into the protocol you can modify the binder computation. Adding data that has
-been authenticated outside the TLS protocol allows the client to add
-authentication guarantees to the TLS handshake beyond those directly provided.
-In particular this can be used to bind external protocols to the TLS protocol.
-For example if two parties have a shared PSK, but have not agreed a hash
-algorithm then this process can be used to ensure agreement on the hash
-algorithm before the TLS handshake completes, and further to ensure that both
-parties agree on the selected hash algorithm.
-
-The change to the binder key computation is necessary because otherwise one
-party may believe it has agreed on some authenticated data, but the other
-party may believe that it is simply using a vanilla PSK, and be unaware of the
-extra context implied.
+To achieve this, pre-shared keys modify the binder key computation. This is
```

Comment:
I'm not sure this is quite sharp enough. A key aspect of the change to the binder key computation is the modification of the labels.